2.0/m_joinpartspam.cpp

```diff
@@ -188,6 +199,21 @@ class ModuleJoinPartSpam : public Module
 		ServerInstance->Modules->SetPriority(this, I_OnUserJoin, PRIORITY_LAST);
 	}
 
+	void OnRehash(User*)
+	{
+		/* Check for a channel redirect value and send an Announcement SNOTICE
+                 * if it's invalid or the ban redirect module is not loaded */
+		ConfigTag* tag = ServerInstance->Config->ConfValue("joinpartspam");
+		const std::string redirchan = tag->getString("redirchan");
+		if (!redirchan.empty())
+		{
+			if (!ServerInstance->IsChannel(redirchan.c_str(), ServerInstance->Config->Limits.ChanMax))
```

Comment:
True, if they configure a channel, they imply they want to use it. What's recommended practice for a live rehash and invalid parameter? 1) Store the old string if there is one, go back to using it; 2) Clear the string and therefore it won't be used later (at the .empty() check pre-ban setting).